src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala

```diff
@@ -147,7 +149,23 @@ class PodsController(
       }
     }
 
-  def status(podId: PathId): Route = ???
+  @SuppressWarnings(Array("all")) // async/await
+  def status(podId: PathId): Route =
+    authenticated.apply { implicit identity =>
+      podManager.find(podId) match {
+        case None =>
+          reject(Rejections.EntityNotFound.noPod(podId))
+        case Some(pod) =>
+          authorized(ViewRunSpec, pod).apply {
+            onSuccess(podStatusService.selectPodStatus(podId)) { maybeStatus =>
+              // If selectPodStatus returns None this is a bug since find(podId) already verifies that the pod exists.
```

Comment:
is it possible that the pod might be deleted by another request in between find and select calls?

Reply:
I think so. However, IMHO that is still a bug in our setup. It's actually an issue with `selectPodStatus` which does not distinguish between pod not found or access rejected.